Classes/RealUrl/TitleMapping.php

```diff
@@ -0,0 +1,233 @@
+<?php
+declare(strict_types=1);
+namespace JWeiland\ServiceBw2\RealUrl;
+
+/*
+ * This file is part of the service_bw2 project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+use DmitryDulepov\Realurl\Decoder\UrlDecoder;
+use DmitryDulepov\Realurl\EncodeDecoderBase;
+use DmitryDulepov\Realurl\Encoder\UrlEncoder;
+use DmitryDulepov\Realurl\Utility;
+use JWeiland\ServiceBw2\Domain\Repository\AbstractRepository;
+use JWeiland\ServiceBw2\Domain\Repository\LebenslagenRepository;
+use JWeiland\ServiceBw2\Domain\Repository\LeistungenRepository;
+use JWeiland\ServiceBw2\Domain\Repository\OrganisationseinheitenRepository;
+use TYPO3\CMS\Core\Resource\Exception\InvalidPathException;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Object\ObjectManager;
+
+/**
+ * Class TitleMapping
+ *
+ * @package JWeiland\ServiceBw2\RealUrl
+ */
+class TitleMapping
+{
+    /**
+     * @var ObjectManager
+     */
+    protected $objectManager;
+
+    /**
+     * @var UrlEncoder
+     */
+    protected $urlEncoder;
```

Comment:
You need it only ones and you do not fill it while constructor, so initiate it where you need it directly and remove it. Same with urlDecoder